public/app/plugins/panel/state-timeline/utils.ts

```diff
@@ -401,6 +401,11 @@ export function prepareTimelineFields(
       refFieldPseudoMax: timeRange.to.valueOf(),
     });
 
+    // Mark the field state as having a null threhold applied
+    if (frame.fields[0].state !== null && frame.fields[0].state !== undefined) {
+      frame.fields[0].state.nullThresholdApplied = true;
+    }
+
```

Comment:
i think generally the pattern is to skip the `if` and do this:

```js
frame.fields[0].state = {
  ...frame.fields[0].state,
  nullThresholdApplied: true,
}
```

Reply:
Agreed! I made the suggested change.